grails-app/controllers/org/bbop/apollo/UserController.groovy

```diff
@@ -24,98 +24,120 @@ class UserController {
+                List<Long> organismsWithPermissions = new ArrayList<>()
+                log.debug "number of groups for user: ${userOrganismPermissionList3?.size()} for ${it.username}"
+                for (UserOrganismPermission userOrganismPermission in userOrganismPermissionList3) {
+                    if (userOrganismPermission.organism in allowableOrganisms) {
+                        JSONObject organismJSON = new JSONObject()
+                        organismJSON.put("organism", userOrganismPermission.organism.commonName)
+                        organismJSON.put("permissions", userOrganismPermission.permissions)
+                        organismJSON.put("userId", userOrganismPermission.userId)
+                        organismJSON.put("id", userOrganismPermission.id)
+                        organismPermissionsArray.add(organismJSON)
+                        organismsWithPermissions.add(userOrganismPermission.organism.id)
+                    }
+                }
+
+                // if an organism has permissions
+                Set<Organism> organismList = allowableOrganisms.findAll() {
+                    !organismsWithPermissions.contains(it.id)
+                }
+
+                for (Organism organism in organismList) {
                     JSONObject organismJSON = new JSONObject()
-                    organismJSON.put("organism", userOrganismPermission.organism.commonName)
-                    organismJSON.put("permissions", userOrganismPermission.permissions)
-                    organismJSON.put("userId", userOrganismPermission.userId)
-                    organismJSON.put("id", userOrganismPermission.id)
+                    organismJSON.put("organism", organism.commonName)
+                    organismJSON.put("permissions", "[]")
+                    organismJSON.put("userId", it.id)
                     organismPermissionsArray.add(organismJSON)
-                    organismsWithPermissions.add(userOrganismPermission.organism.id)
                 }
-            }
 
-            // if an organism has permissions
-            Set<Organism> organismList = allowableOrganisms.findAll() {
-                !organismsWithPermissions.contains(it.id)
-            }
-
-            for (Organism organism in organismList) {
-                JSONObject organismJSON = new JSONObject()
-                organismJSON.put("organism", organism.commonName)
-                organismJSON.put("permissions", "[]")
-                organismJSON.put("userId", it.id)
-                organismPermissionsArray.add(organismJSON)
-            }
 
+                userObject.organismPermissions = organismPermissionsArray
 
-            userObject.organismPermissions = organismPermissionsArray
+                returnArray.put(userObject)
+            }
 
-            returnArray.put(userObject)
+            render returnArray as JSON
+        }
+        catch(Exception e) {
+            response.status=500
```

Comment:
My only comment is that this should be:

        response.status = HttpStatus.SC_INTERNAL_SERVER_ERROR